ddtrace/internal/ci_visibility/writer.py

```diff
@@ -65,8 +66,12 @@ def __init__(
         if not intake_url:
             intake_url = "%s.%s" % (AGENTLESS_BASE_URL, AGENTLESS_DEFAULT_SITE)
 
-        client = CIVisibilityProxiedEventClient() if use_evp else CIVisibilityAgentlessEventClient()
         headers = headers or dict()
+        headers["dd-api-key"] = os.environ.get("DD_API_KEY") or ""
+        if not headers["dd-api-key"]:
+            raise ValueError("Required environment variable DD_API_KEY not defined")
```

Comment:
We don't want this in EVP Proxy mode, though.

Reply:
yeah good catch, I saw this when running e2e tests